examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js

```diff
@@ -0,0 +1,655 @@
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/816'
+          },
+          {
+            codecConfiguration: codecConfigurationH264At544p,
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/544'
+          },          {
+            codecConfiguration: codecConfigurationH264At272p,
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/272'
+          }
+        ];
+
+        const promiseMap = Promise.map(streamDefinitions, (streamDefinition) => {
+          return createStreamWithMuxingsAndDRMsAndManifestResources(
+            streamDefinition.codecConfiguration, input, output, streamDefinition.outputPathDiscriminator, encoding,
+            createdDashManifest, createdDashManifestPeriod, streamDefinition.adaptationSet, hlsManifest
+          );
+        }, {concurrency: 1});
+
+        promiseMap.then((result) => {
+          console.log('Successfully created all resources. Starting Encoding Process...');
+          startEncodingAndWaitForItToBeFinished(encoding).then(() => {
+            console.log('Successfully Finished Encoding Process.. Starting manifest creation...');
+            const dashCreationPromise = startDashManifestCreationAndWaitForItToBeFinished(createdDashManifest);
+            const hlsCreationPromise = startHlsManifestCreationAndWaitForItToBeFinished(hlsManifest);
+            Promise.all([dashCreationPromise, hlsCreationPromise]).then(() => {
+              console.log('Successfully finished DASH and HLS manifest creation!');
+              resolve(result);
+            });
+          });
+        }).catch((error) => {
+          console.error('An error occured: ', error);
+          reject(error);
+        });
+      });
+    });
+  });
+};
+
+const createInput = () => {
+  return new Promise((resolve, reject) => {
+    const httpInput = {
+      name: 'My awesome HTTP Input',
+      description: 'Little description for my awesome HTTP Input',
+      host: INPUT_HTTP_HOST
+    };
+    bitmovin.encoding.inputs.http.create(httpInput).then((createdInput) => {
+      console.log('Successfully created HTTP Input.', createdInput);
+      resolve(createdInput);
+    }).catch((error) => {
+      console.error('Unable to create HTTP Input.', error);
+      reject(error);
+    })
+  });
+};
+
+const createOutput = () => {
+  return new Promise((resolve, reject) => {
+    const s3Output = {
+      name: 'My awesome S3 Output',
+      description: 'Little description for my awesome S3 Output',
+      accessKey: OUTPUT_S3_ACCESS_KEY,
+      secretKey: OUTPUT_S3_SECRET_KEY,
+      bucketName: OUTPUT_S3_BUCKET_NAME
+    };
+    bitmovin.encoding.outputs.s3.create(s3Output).then((createdOutput) => {
+      console.log('Successfully created S3 Output.', createdOutput);
+      resolve(createdOutput);
+    }).catch((error) => {
```

Comment:
We could cut down on all the error handling with a function like this:

```
const logErrorPromise = (errorMessage) => {
	return (value) => {
  	console.log(errorMessage, value);
    return value;
  };
};
```

Then we would rewrite this as 
`.catch(logErrorPromise('nable to create S3 Output.'));`

Try it  here: https://jsfiddle.net/39wryt4b/